src/Core/Entities/AccessPolicy.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Bit.Core.Utilities;
+
+namespace Bit.Core.Entities;
+
+public class AccessPolicy : ITableObject<Guid>
+{
+    public Guid Id { get; set; }
+
+    // Object to grant access from
+    public Guid? OrganizationUserId { get; set; }
+    public Guid? GroupId { get; set; }
+    public Guid? ServiceAccountId { get; set; }
+
+    // Object to grant access to
+    public Guid? GrantedProjectId { get; set; }
+    public Guid? GrantedServiceAccountId { get; set; }
+
+    // Access
+    public bool Read { get; set; }
+    public bool Write { get; set; }
+
+    public DateTime CreationDate { get; set; }
+    public DateTime RevisionDate { get; set; }
+
+    public void SetNewId()
+    {
+        Id = CoreHelpers.GenerateComb();
+    }
+}
+
+public abstract class BaseAccessPolicy
+{
+    public Guid Id { get; set; }
+
+    // Access
+    public bool Read { get; set; }
+    public bool Write { get; set; }
+
+    public DateTime CreationDate { get; set; }
+    public DateTime RevisionDate { get; set; }
+
+    public void SetNewId()
+    {
+        Id = CoreHelpers.GenerateComb();
+    }
+}
+
+public class UserProjectAccessPolicy : BaseAccessPolicy
+{
+    public Guid? OrganizationUserId { get; set; }
+    public Guid? GrantedProjectId { get; set; }
+}
```

Comment:
This feels like a weird way to do it. I had expected `AccessPolicy` to be an interface that was explicitly implemented in each of the `___AccessPolicy` types to hide irrelevant properties

```suggestion
public interface IAccessPolicy : ITableObject<Guid>
{
    Guid Id { get; set; }

    // Object to grant access from
    Guid? OrganizationUserId { get; set; }
    Guid? GroupId { get; set; }
    Guid? ServiceAccountId { get; set; }

    // Object to grant access to
    Guid? GrantedProjectId { get; set; }
    Guid? GrantedServiceAccountId { get; set; }

    // Access
    bool Read { get; set; }
    bool Write { get; set; }

    DateTime CreationDate { get; set; }
    DateTime RevisionDate { get; set; }

    void SetNewId();
}

public class UserProjectAccessPolicy : IAccessPolicy
{
    public Guid? OrganizationUserId { get; set; }
    public Guid? GrantedProjectId { get; set; }

    public bool Read { get; set; }
    public bool Write { get; set; }

    Guid IAccessPolicy.GroupId { get; set; }
    Guid IAccessPolicy.ServiceAccountId { get; set; }
    Guid IAccessPolicy.GrantedServiceAccountId { get; set; }

    public void SetNewId()
    {
        Id = CoreHelpers.GenerateComb();
    }
}
```

Reply:
I don't think that will play well with EF and I don't particularly feel like digging into potentially finding a solution for it :(.